src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -27,7 +30,12 @@ namespace
     std::mutex g_lib_lock;
     std::weak_ptr<coreclr_t> g_lib_coreclr;
 
-    int create_coreclr(const hostpolicy_context_t &context, host_mode_t mode, std::unique_ptr<coreclr_t> &coreclr)
+    std::mutex g_context_lock;
+    std::condition_variable g_context_cv;
+    std::unique_ptr<hostpolicy_context_t> g_context;
+    std::atomic<bool> g_context_initializing(false);
```

Comment:
Just like in hostfxr - can you please add comment describing the synchronization design and what are the various globals for?